pkg/util/docker/docker.go

```diff
@@ -227,13 +227,36 @@ type Container struct {
 	cgroup *ContainerCgroup
 }
 
+// Inspect allows getting the full docker inspect of a Container
 func (c *Container) Inspect(withSize bool) (types.ContainerJSON, error) {
-	if globalDockerUtil == nil {
-		return types.ContainerJSON{}, fmt.Errorf("DockerUtil not initialized")
+	cj, err := Inspect(c.ID, withSize)
+	return cj, err
+}
+
+// Inspect returns a docker inspect object for a given container ID.
+// It tries to locate the container in the inspect cache before making the docker inspect call
+func Inspect(id string, withSize bool) (types.ContainerJSON, error) {
+	cacheKey := GetInspectCacheKey(id)
+	var container types.ContainerJSON
+	var err error
+	var ok bool
+
+	if cached, hit := cache.Cache.Get(cacheKey); hit {
+		container, ok = cached.(types.ContainerJSON)
+		if !ok {
+			log.Errorf("invalid cache format, forcing a cache miss")
+		}
+	} else {
+		if globalDockerUtil == nil {
+			return types.ContainerJSON{}, fmt.Errorf("DockerUtil not initialized")
+		}
+		container, _, err = globalDockerUtil.cli.ContainerInspectWithRaw(context.Background(), id, withSize)
 	}
 
-	res, _, err := globalDockerUtil.cli.ContainerInspectWithRaw(context.Background(), c.ID, withSize)
-	return res, err
+	// cache the inspect for 10 seconds to reduce pressure on the daemon
+	cache.Cache.Set(cacheKey, container, 10*time.Second)
```

Comment:
This line should be in the else no? Otherwise you refresh the cache even if you hit it